app/controllers/api/subcollections/vms.rb

```diff
@@ -23,16 +23,31 @@ def vms_query_resource(object)
 
       private
 
+      def decorator_selection
+        params['decorators'].to_s.split(",")
+      end
+
+      def decorator_selection_for(collection)
+        decorator_selection.collect do |attr|
+          /\A#{collection}\.(?<name>.*)\z/.match(attr) { |m| m[:name] }
+        end.compact
+      end
+
       def create_vm_attributes_hash(vm_attrs, vm)
         vm_attrs.each_with_object({}) do |attr, hash|
           hash[attr] = vm.public_send(attr.to_sym) if vm.respond_to?(attr.to_sym)
         end.compact
       end
 
       def create_vm_decorators_hash(vm_decorators, vm)
-        vm_decorators.each_with_object({}) do |name, hash|
-          hash[name] = vm.decorate.public_send(name.to_sym) if vm.decorate.respond_to?(name.to_sym)
-        end.compact
+        hash = {}
+        if vm_decorators.include? 'supports_console?'
+          hash['supports_console?'] = vm.supports_console?
+        end
+        if vm_decorators.include? 'supports_cockpit?'
+          hash['supports_cockpit?'] = vm.supports_launch_cockpit?
+        end
+        hash
```

Comment:
@abellotti My impression was that people want decorators *gone* from the API .. that's why I took this path, we still need to support the ones that were used, but the idea of this PR is that those would be the only ones. (And can be replaced/deprecated in a future API version.)

Does your objection still stand given that?